### DISCUSSION TITLE
Does Valent use a different port than GSConnect does?

### DISCUSSION THREAD

**Discussion by schuelermine** [andyholmes/valent#424] [Help] [answered]
I’m connecting my device (Pixel 6 running GrapheneOS) via USB tethering and am unable to connect. I have my firewall whitelist ports 1714-1764 for KDE Connect, but it doesn’t seem to work.

**Comment by andyholmes** [comment, marked as answer]
Yes, Valent uses port `1716` for TCP & UDP, and `1739-1764` for auxiliary TCP streams. Technically I think "we" reserve `1716-1764` for TCP/UDP, but that's what actually gets used. `1714` was the default port some years back, but isn't used anymore.

Note that Valent will log a warning, but won't show a notification if it can't own port `1716` (i.e. GSConnect or KDE Connect are running already). Usually `lsof -i :1716` will show the culprit, or I've found the [What IP](https://flathub.org/apps/details/org.gabmus.whatip) app an easy tool for checking network behviour.

Let me know! I've been working on in-app help for Valent, so it would be nice to know where the breaking points are :)

**Reply by schuelermine** [reply]
Thank you for the info. My problem seems to have magically fixed itself in the meantime.

**Comment by colemickens** [comment]
Hi @andyholmes ! I don't want to open a new issue but I will make an explicit "help" recommendation/request:

a warning, especially with no devices connected, about opening ports would be great. Although, it would be great if this also wasn't necessary, though I'm sure that's out of scope for `valent`.

**Reply by andyholmes** [reply]
No problem :)

At the moment, there's no infrastructure in place for plugins to report errors to the user (and everything is a plugin in Valent). There will probably be something subtle, but not like with GSConnect. To be honest, we had a notification marked urgent which linked to clear instructions on how to troubleshoot and resolve, but it really didn't make a difference.

Generally one of two things happen:

1. The user will start grasping at straws and install as many KDE Connect clients as they can, making the problem...well the same, actually. But definitely not helping :)
2. There's actually a bug (rare, but your case counts), they're doing something funky with their network (VPNs, proxy, tunneling, bridged routers, etc), or they have some firewall the user docs just don't cover.

In the first case, it ends up they just ignored the notification or figured the problem couldn't possibly be another KDE Connect client (spoiler alert: always is).

In the second case, like yours, it just can't be fixed by the user anyways and we have to manually troubleshoot or figure out what new firewall thingy they're using.

I have been following along the UI with some in-app help documentation, so maybe the ignorable notification can be skipped and the in-app troubleshooting will be less likely to get ignored:

![Screenshot from 2023-04-03 20-16-45](https://user-images.githubusercontent.com/1265208/229678160-a604e525-0598-45d5-b17c-33f62681d1ca.png)
